clients/include/testing_gels_batched.hpp

```diff
@@ -0,0 +1,195 @@
+    host_vector<T>       info_res(batchCount);
+    host_vector<T>       info(batchCount);
+    int                  info_input(-1);
+
+    device_batch_vector<T> dA(A_size, 1, batchCount);
+    device_batch_vector<T> dB(B_size, 1, batchCount);
+    device_vector<int>     dInfo(batchCount);
+
+    double             gpu_time_used, hipblas_error;
+    hipblasLocalHandle handle(argus);
+
+    // Initial hA, hB, hX on CPU
+    hipblas_init<T>(hA, true);
+    hipblas_init<T>(hB);
+    hB_res.copy_from(hB);
+
+    // scale A to avoid singularities
+    for(int b = 0; b < batchCount; b++)
+    {
+        for(int i = 0; i < N; i++)
+        {
+            for(int j = 0; j < N; j++)
+            {
+                if(i == j)
+                    hA[b][i + j * lda] += 400;
+                else
+                    hA[b][i + j * lda] -= 4;
+            }
+        }
+    }
+
+    // Copy data from CPU to device
+    CHECK_HIP_ERROR(dA.transfer_from(hA));
+    CHECK_HIP_ERROR(dB.transfer_from(hB));
+
+    if(argus.unit_check || argus.norm_check)
+    {
+        /* =====================================================================
+            HIPBLAS
+        =================================================================== */
+        CHECK_HIPBLAS_ERROR(hipblasGelsBatchedFn(handle,
+                                                 trans,
+                                                 M,
+                                                 N,
+                                                 nrhs,
+                                                 dA.ptr_on_device(),
+                                                 lda,
+                                                 dB.ptr_on_device(),
+                                                 ldb,
+                                                 &info_input,
+                                                 dInfo,
+                                                 batchCount));
+
+        // copy output from device to CPU
+        CHECK_HIP_ERROR(hB_res.transfer_from(dB));
+        CHECK_HIP_ERROR(
+            hipMemcpy(info_res.data(), dInfo, sizeof(int) * batchCount, hipMemcpyDeviceToHost));
+
+        /* =====================================================================
+           CPU LAPACK
+        =================================================================== */
+        int            sizeW = std::max(1, std::min(M, N) + std::max(std::min(M, N), nrhs));
+        host_vector<T> hW(sizeW);
+
+        for(int b = 0; b < batchCount; b++)
+        {
+            info[b] = cblas_gels(transc, M, N, nrhs, hA[b], lda, hB[b], ldb, hW.data(), sizeW);
+        }
+
+        hipblas_error
+            = norm_check_general<T>('F', std::max(M, N), nrhs, ldb, hB, hB_res, batchCount);
+
+        for(int b = 0; b < batchCount; b++)
+        {
+            if(info_res[b] != 0)
+                hipblas_error += 1.0;
+            if(info[b] != info_input)
+                hipblas_error += 1.0;
```

Comment:
I missed this on the first PR, but I think it makes more sense to compare `info[b]` with `info_res[b]`. If the matrix is singular, `info[b]` and `info_res[b]` will both be non-zero while `info_input` should be zero.